src/smt/set_defaults.cpp

```diff
@@ -289,6 +289,14 @@ void setDefaults(LogicInfo& logic, bool isInternalSubsolver)
     disableProofs = true;
   }
 
+  if (options::unsatCores()
```

Comment:
This isn't going to quite work I don't think, since this disables *all* proof production, including the allocation of proof node manager. We also need to keep proof production in the preprocessor.

I think I would recommend deleting this block of code and then catching what we need to do in smt::ProofManager.

Reply:
Oh yeah, I was stupid... We should probably just issue a warning that the user can't produce proofs, but of course the code should not be disabled...